### PR TITLE
Refine docs: troubleshoot lock conflicts 

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8191
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: pessimistic-transaction.md [hunk 1]
```diff
@@ -139,6 +139,20 @@ TiDB supports the following two isolation levels in the pessimistic transaction
...
+## Pessimistic transaction commit process
+
+The commit phase of the pessimistic transaction mode and the optimistic transaction mode in TiDB has the same logic, and both commits are in the 2PC mode. The important adaptation of pessimistic transactions is DML execution.
+
+![TiDB pessimistic transaction commit logic](/media/pessimistic-transaction-commit.png)
+
+The pessimistic transaction adds an `Acquire Pessimistic Lock` phase before 2PC. This phase includes the following steps:
+
+1. (Same as the optimistic transaction mode) Receive the `begin` request from the client, and the current timestamp is this transaction's start_ts.
+2. When the TiDB server receives an `update` request from the client, the TiDB server initiates a pessimistic lock request to the TiKV server, and the lock is persisted to the TiKV server.
```
Comment (MyonKeminta) [inline pessimistic-transaction.md:151]: We should be talking about "writing new data", rather than talking about `update` statements specifically.

**Review by MyonKeminta — COMMENTED** [review]
Rest LGTM

File: pessimistic-transaction.md (hunk 1 above)
Comment (qiancai) [inline pessimistic-transaction.md:151]:
```suggestion
2. When the TiDB server receives a request of writing new data from the client, the TiDB server initiates a pessimistic lock request to the TiKV server, and the lock is persisted to the TiKV server.
```

Comment (qiancai) [reply pessimistic-transaction.md:151]: @MyonKeminta Please check if the following update is ok. Thanks.

image

File: pessimistic-transaction.md [hunk 2]
```diff
@@ -139,6 +139,20 @@ TiDB supports the following two isolation levels in the pessimistic transaction
 
 - [Read Committed](/transaction-isolation-levels.md#read-committed-isolation-level). You can set this isolation level using the [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) statement.
 
+## Pessimistic transaction commit process
+
+The commits of the pessimistic transactions and the optimistic transactions in TiDB have the same logic. Both transactions adopt the two-phase commit (2PC) mode. The important adaptation of pessimistic transactions is DML execution.
+
+![TiDB pessimistic transaction commit logic](/media/pessimistic-transaction-commit.png)
```
Comment (qiancai) [inline pessimistic-transaction.md:146]:
```suggestion
![TiDB pessimistic transaction commit process](/media/pessimistic-transaction-commit.png)
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: pessimistic-transaction.md [hunk 3]
```diff
@@ -139,6 +139,20 @@ TiDB supports the following two isolation levels in the pessimistic transaction
 
 - [Read Committed](/transaction-isolation-levels.md#read-committed-isolation-level). You can set this isolation level using the [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) statement.
 
+## Pessimistic transaction commit process
+
+The commit phase of the pessimistic transaction mode and the optimistic transaction mode in TiDB has the same logic, and both commits are in the 2PC mode. The important adaptation of pessimistic transactions is DML execution.
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply pessimistic-transaction.md]: Changing "the commit phase" to "commits" doesn't make sense to me. Here it's supposed to stands for the *commit* operation of a transaction. What we want to say here is 悲观事务和乐观事务的*提交阶段*逻辑相同.

File: pessimistic-transaction.md [hunk 4]
```diff
@@ -139,6 +139,20 @@ TiDB supports the following two isolation levels in the pessimistic transaction
...
+## Pessimistic transaction commit process
+
+The commits of the pessimistic transactions and the optimistic transactions in TiDB have the same logic. Both transactions adopt the two-phase commit (2PC) mode. The important adaptation of pessimistic transactions is DML execution.
+
+![TiDB pessimistic transaction commit process](/media/pessimistic-transaction-commit.png)
+
+The pessimistic transaction adds an `Acquire Pessimistic Lock` phase before 2PC. This phase includes the following steps:
+
+1. (Same as the optimistic transaction mode) TiDB receives the `begin` request from the client, and the current timestamp is this transaction's start_ts.
+2. When the TiDB server receives a request of writing new data from the client, the TiDB server initiates a pessimistic lock request to the TiKV server, and the lock is persisted to the TiKV server.
```
Comment (MyonKeminta) [inline pessimistic-transaction.md:151]:
```suggestion
2. When the TiDB server receives a writing request from the client, the TiDB server initiates a pessimistic lock request to the TiKV server, and the lock is persisted to the TiKV server.
```

**Review by MyonKeminta — COMMENTED** [review]
Rest LGTM

File: pessimistic-transaction.md [hunk 5]
```diff
@@ -139,6 +139,20 @@ TiDB supports the following two isolation levels in the pessimistic transaction
 
 - [Read Committed](/transaction-isolation-levels.md#read-committed-isolation-level). You can set this isolation level using the [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) statement.
 
+## Pessimistic transaction commit process
+
+The commits of the pessimistic transactions and the optimistic transactions in TiDB have the same logic. Both transactions adopt the two-phase commit (2PC) mode. The important adaptation of pessimistic transactions is DML execution.
```
Comment (qiancai) [inline pessimistic-transaction.md:144]:
```suggestion
The logic of TiDB pessimistic transactions and optimistic transactions in the commit phase is the same. Both transactions adopt the two-phase commit (2PC) mode. The important adaptation of pessimistic transactions is DML execution.
```

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**